From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sat, 18 May 2019 19:45:10 +0200
Subject: [PATCH] {MAYBE UPS} drivers/staging/kpc2000/kpc_dma/fileops: fix
 printf format
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gcc on x86-32 reports:

    drivers/staging/kpc2000/kpc_dma/fileops.c:58:35: error: format ‘%ld’
    expects argument of type ‘long int’, but argument 7 has type
    ‘size_t’ {aka ‘unsigned int’} [-Werror=format=]
      dev_dbg(&priv->ldev->pldev->dev, "kpc_dma_transfer(priv = [%p], kcb = [%p], iov_base = [%p], iov_len = %ld) ldev = [%p]\n", priv, kcb, (void*)iov_base, iov_len, ldev);
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    include/linux/dynamic_debug.h:125:15: note: in definition of macro ‘__dynamic_func_call’
       func(&id, ##__VA_ARGS__);  \
                   ^~~~~~~~~~~
    include/linux/dynamic_debug.h:157:2: note: in expansion of macro ‘_dynamic_func_call’
      _dynamic_func_call(fmt,__dynamic_dev_dbg,   \
      ^~~~~~~~~~~~~~~~~~
    include/linux/device.h:1493:2: note: in expansion of macro ‘dynamic_dev_dbg’
      dynamic_dev_dbg(dev, dev_fmt(fmt), ##__VA_ARGS__)
      ^~~~~~~~~~~~~~~
    include/linux/device.h:1493:23: note: in expansion of macro ‘dev_fmt’
      dynamic_dev_dbg(dev, dev_fmt(fmt), ##__VA_ARGS__)
                           ^~~~~~~
    drivers/staging/kpc2000/kpc_dma/fileops.c:58:2: note: in expansion of macro ‘dev_dbg’
      dev_dbg(&priv->ldev->pldev->dev, "kpc_dma_transfer(priv = [%p], kcb = [%p], iov_base = [%p], iov_len = %ld) ldev = [%p]\n", priv, kcb, (void*)iov_base, iov_len, ldev);
      ^~~~~~~
---
 drivers/staging/kpc2000/kpc2000/core.c    | 2 +-
 drivers/staging/kpc2000/kpc_dma/fileops.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/kpc2000/kpc2000/core.c b/drivers/staging/kpc2000/kpc2000/core.c
index 40390cdd3c8d..e7fc420497b8 100644
--- a/drivers/staging/kpc2000/kpc2000/core.c
+++ b/drivers/staging/kpc2000/kpc2000/core.c
@@ -146,7 +146,7 @@ static int  read_system_regs(struct kp2000_device *pcard)
     
     read_val = readq(pcard->sysinfo_regs_base + REG_MAGIC_NUMBER);
     if (read_val != KP2000_MAGIC_VALUE){
-        dev_err(&pcard->pdev->dev, "Invalid magic!  Got: 0x%016llx  Want: 0x%016lx\n", read_val, KP2000_MAGIC_VALUE);
+        dev_err(&pcard->pdev->dev, "Invalid magic!  Got: 0x%016llx  Want: 0x%016llx\n", read_val, (u64)KP2000_MAGIC_VALUE);
         return -EILSEQ;
     }
     
diff --git a/drivers/staging/kpc2000/kpc_dma/fileops.c b/drivers/staging/kpc2000/kpc_dma/fileops.c
index 5741d2b49a7d..be074f91153f 100644
--- a/drivers/staging/kpc2000/kpc_dma/fileops.c
+++ b/drivers/staging/kpc2000/kpc_dma/fileops.c
@@ -55,7 +55,7 @@ int  kpc_dma_transfer(struct dev_private_data *priv, struct kiocb *kcb, unsigned
 	ldev = priv->ldev;
 	BUG_ON(ldev == NULL);
 	
-	dev_dbg(&priv->ldev->pldev->dev, "kpc_dma_transfer(priv = [%p], kcb = [%p], iov_base = [%p], iov_len = %ld) ldev = [%p]\n", priv, kcb, (void*)iov_base, iov_len, ldev);
+	dev_dbg(&priv->ldev->pldev->dev, "kpc_dma_transfer(priv = [%p], kcb = [%p], iov_base = [%p], iov_len = %ld) ldev = [%p]\n", priv, kcb, (void*)iov_base, (unsigned long)iov_len, ldev);
 	
 	acd = (struct aio_cb_data *) kzalloc(sizeof(struct aio_cb_data), GFP_KERNEL);
 	if (!acd){
-- 
